core/frontend/src/components/vehiclesetup/configuration/lights.vue

```diff
@@ -0,0 +1,156 @@
+  name: 'LightsConfigration',
+  data() {
+    return {
+      lights1_new_param: undefined as (string | undefined), // item-value must be a primitive
+      lights2_new_param: undefined as (string | undefined),
+      steps_new_value: 0,
+    }
+  },
+  computed: {
+    light_steps(): Parameter | undefined {
+      return autopilot_data.parameter('JS_LIGHTS_STEPS')
+    },
+    recommended_params(): Parameter[] {
+      // return parameters in servo_params that are on channel 9 and higher
+      return this.servo_params.filter((param) => {
+        const servoNumber = parseInt(param.name.replace('SERVO', '').split('_')[0], 10)
+        return servoNumber >= 9
+      })
+    },
+    servo_params(): Parameter[] {
+      return autopilot_data.parameterRegex('SERVO[0-9]+_FUNCTION')
+    },
+    lights1_param(): Parameter | undefined {
+      return this.servo_params.filter((param) => param.value === Lights.Lights1)[0]
+    },
+    lights2_param(): Parameter | undefined {
+      return this.servo_params.filter((param) => param.value === Lights.Lights2)[0]
+    },
+  },
+  watch: {
+    steps_param(new_value: Parameter | undefined) {
+      this.steps_new_value = new_value?.value ?? 0
+    },
+    steps_new_value(new_value: number | undefined) {
+      if (new_value) {
+        if (this.light_steps?.value === new_value) {
+          return
+        }
+        mavlink2rest.setParam('JS_LIGHTS_STEPS', new_value, autopilot_data.system_id)
+      }
+    },
+
+    lights1_param(new_value: Parameter | undefined) {
+      this.lights1_new_param = new_value?.name
+    },
+    lights2_param(new_value: Parameter | undefined) {
+      this.lights2_new_param = new_value?.name
+    },
+    // todo: refactor these two methods into one
+    lights1_new_param(new_param_name: string | undefined) {
+      if (new_param_name) {
+        // reset any other parameter using lights1 to undefined
+        for (const old_param of this.servo_params) {
+          if (old_param.name === new_param_name) {
+            continue
+          }
+          if (old_param.value === Lights.Lights1) {
+            // set the old parameter to DISABLED
+            mavlink2rest.setParam(old_param.name, Lights.DISABLED, autopilot_data.system_id)
+          }
+        }
+        mavlink2rest.setParam(new_param_name, Lights.Lights1, autopilot_data.system_id)
+      }
+    },
+    lights2_new_param(new_param_name: string | undefined) {
+      if (new_param_name) {
+        // reset any other parameter using lights2 to undefined
+        for (const old_param of this.servo_params) {
+          if (old_param.name === new_param_name) {
+            continue
+          }
+          if (old_param.value === Lights.Lights2) {
+            // set the old parameter to DISABLED
+            mavlink2rest.setParam(old_param.name, Lights.DISABLED, autopilot_data.system_id)
+          }
+        }
+        mavlink2rest.setParam(new_param_name, Lights.Lights2, autopilot_data.system_id)
+      }
```

Comment:
Can we encapsulate this in a single function for both lights ?

Reply:
done